### PR TITLE
Add ApprovedAt to Collectives

### PR DESCRIPTION
**Pull request by flickz**
This PR attempt to fix [#1878](https://github.com/opencollective/opencollective/issues/1878) && [#2100](https://github.com/opencollective/opencollective/issues/2100).

- Adds `approvedAt` to Collective table
- Add migration file to set the value of `collective.approvedAt` to colllective host member `since` date.
- Update the value of `approvedAt` when collective is approved, change host or archived.
- Add `isApproved` property to `Collective` query.
- Remove collective host in `archiveCollective` mutation.

### CODE REVIEW

**Comment by znarf** [comment]
We should avoid complex behavior in migrations (20190805114605-set-approved-collective-approvedAt-column-value.js). We should ideally stick to pure SQL.

- In `20190805102047-add-approvedAt-column-to-collective.js`, can you add `approvedAt` to a default value like today for approved collectives?
- Can you move `20190805114605-set-approved-collective-approvedAt-column-value.js` in  `scripts` and adjust the syntax so that it's not a mutation anymore? We will trigger it manually.

**Comment by flickz** [comment]
Okay, let me adjust that.

File: server/graphql/v1/mutations/collectives.js [hunk 1]
```diff
@@ -634,7 +634,19 @@ export async function archiveCollective(_, args, req) {
...
+    where: {
+      CollectiveId: collective.id,
+      MemberCollectiveId: collective.HostCollectiveId,
+      role: roles.HOST,
+    },
+  });
+
+  if (membership) {
+    membership.destroy();
+  }
```
Comment (znarf) [inline server/graphql/v1/mutations/collectives.js]: Thank you for adding that. I think we need to also add `HostCollectiveId: null`in `collective.update`.

Comment (flickz) [reply server/graphql/v1/mutations/collectives.js]: Yes we do, I just added it.

File: migrations/20190805102047-add-approvedAt-column-to-collective.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    const colParms = {
+      type: Sequelize.DATE,
+    };
+    await queryInterface.addColumn('Collectives', colName, colParms);
+    await queryInterface.addColumn('CollectiveHistories', colName, colParms);
+    return queryInterface.sequelize.query(`
+      UPDATE "Collectives" c
+        SET "approvedAt" = '2019-08-06'
```
Comment (znarf) [inline migrations/20190805102047-add-approvedAt-column-to-collective.js]: Sorry, also missed that until now. I think we should limit the update to `type=COLLECTIVE`. Also `HostCollectiveId IS NOT NULL`to be on the safe side.

Comment (flickz) [reply migrations/20190805102047-add-approvedAt-column-to-collective.js]: Okay.

**Pull request merged**